mesheryctl/internal/cli/root/system/context/create.go

```diff
@@ -23,35 +20,25 @@ var createContextCmd = &cobra.Command{
 	Long:  `Add a new context to Meshery config.yaml file`,
 	Args:  cobra.ExactArgs(1),
 	RunE: func(cmd *cobra.Command, args []string) error {
-		err := viper.Unmarshal(&configuration)
+		tempContext := utils.TemplateContext
+
+		err := utils.ValidateURL(url)
 		if err != nil {
 			return err
 		}
-		_, exists := configuration.Contexts[args[0]]
-		if exists {
-			return errors.New("error adding context: a context with same name already exists")
-		}
-		ParsedURL, err := urlLib.ParseRequestURI(url)
+		tempContext.Endpoint = url
+
+		err = utils.AddContextToConfig(args[0], tempContext, viper.ConfigFileUsed(), set)
 		if err != nil {
 			return err
 		}
-		if ParsedURL.Scheme != "http" && ParsedURL.Scheme != "https" {
-			return fmt.Errorf("%s is not a supported request scheme", ParsedURL.Scheme)
-		}
-		listOfAdapters := []string{"meshery-istio", "meshery-linkerd", "meshery-consul", "meshery-octarine", "meshery-nsm", "meshery-kuma", "meshery-cpx", "meshery-osm", "meshery-nginx-sm"}
-		configuration.Contexts[args[0]] = models.Context{Endpoint: url, Adapters: listOfAdapters}
-		configuration.CurrentContext = args[0]
-		viper.Set("contexts", configuration.Contexts)
-		viper.Set("current-context", configuration.CurrentContext)
-		err = viper.WriteConfig()
-		if err != nil {
-			log.Fatal(err)
-		}
-		log.Printf("Added %s context and set as current context", args[0])
+
+		log.Printf("Added %s context", args[0])
 		return nil
 	},
 }
 
 func init() {
 	createContextCmd.Flags().StringVarP(&url, "url", "u", "https://localhost:9081", "Meshery Server URL with Port")
```

Comment:
Yikes. What's this reference to localhost here?

Reply:
If you dont pass it a url, it uses that.Also it acts as example when requesting help on cli

Reply:
Ok